### ISSUE TITLE
Loading VS Code from the Mac dock always starts the integrated terminal with bash

### ISSUE THREAD

**Issue by emmahsax** [microsoft/vscode#125898]
Does this issue occur when all extensions are disabled?: Yes/No

Yes

- VS Code Version: 1.56.2
- OS Version: OSX 11.4 (Big Sur)

Essentially, using `terminal.integrated.profiles.osx` and `terminal.integrated.defaultProfile.osx` doesn't work as expected.

### Steps to Reproduce

1. Set the default shell for integrated terminal on OSX to be `zsh` by adding something like the following to the `settings.json` file:
```json
"terminal.integrated.defaultProfile.osx": "zsh",
"terminal.integrated.profiles.osx": {
    "zsh": {
        "path": "/bin/zsh",
        "args": ["-l"]
    },
    "fish": {
        "path": "fish"
    },
    "tmux": {
        "path": "tmux",
        "icon": "terminal-tmux"
    },
    "pwsh": {
        "path": "pwsh",
        "icon": "terminal-powershell"
    }
}
```
2. With VS Code window open, open up a new terminal (it'll probably be `zsh`)
  a. Notice that running `echo $SHELL` outputs `/bin/bash`, not `/bin/zsh`, as would be expected
1. Quit all windows and tabs of VS Code by pressing `CMD + q` **without** closing any windows by clicking with the mouse
2. Double-click the VS Code icon from the Dock (or use Spotlight Search to search for `code`, and then press `ENTER`)
3. Notice that the new terminal that's created starts in `bash` automatically
  a. Notice that running `echo $SHELL` outputs `/bin/bash`, which makes sense because the terminal is `bash`
4. Opening a new terminal on top of the old one (or closing the automatic one and making a new one) starts properly in `zsh`
  a. Notice that running `echo $SHELL` outputs `/bin/bash`, not `/bin/zsh`, as would be expected

### Notes

* If I use Terminal.app, then opening a brand new terminal opens in `zsh`, and running `echo $SHELL` outputs `/bin/zsh`
* Opening VS Code from the Terminal.app (via `code .`), successfully opens a new terminal with `zsh`, and `echo $SHELL` outputs `/bin/zsh`
* Adding back in the following lines opens a new terminal in `zsh` as expected, but `echo $SHELL` still returns `/bin/bash`
  ```json
  "terminal.integrated.shell.osx": "/bin/zsh"
  "terminal.integrated.shellArgs.osx": ["-l"],
  ```
* I've checked all other issues the bot mentions below, and none of them match this one

**Comment by meganrogge** [comment]
I believe this is already fixed, can you see if you can reproduce in the [Insiders build](https://code.visualstudio.com/insiders)?

**Comment by emmahsax** [comment]
@meganrogge Yes, the Insiders build seems to have the successful change! Approximately how long would you say until the regular version has that update?

**Comment by meganrogge** [comment]
We released today @emmahsax so 0 hours 😄

**Comment by emmahsax** [comment]
@meganrogge Actually, I just tried again on my other mac computer, and it looks like (with VS Code 1.57.0), it still has this issue where opening the application from the dock opens a `zsh` terminal, but when I do
```
echo $SHELL
> /bin/bash
```

This is more just mysterious and frustrating than anything else... the `zsh` terminal still works as expected with `zsh`.

Both computers have fresh installations of VS Code with 1.57.0, so the only difference is that one is on Big Sur 11.4 and the other is on Catalina 10.15.7.

**Comment by meganrogge** [comment]
When I do echo $SHELL, I see /bin/zsh

**Comment by meganrogge** [comment]
Oh okay just think I realized the issue - right now, shell/shellArgs take the priority over defaultProfile. So remove those and it should work as you expect.

**Comment by meganrogge** [comment]
Could you please provide me with all of your settings?

**Comment by meganrogge** [comment]
Can you also run `ls -la $(which zsh)` and provide me with the output of that?

**Comment by emmahsax** [comment]
```
% ls -la $(which zsh)
-rwxr-xr-x  1 root  wheel  637840 Sep 21  2020 /bin/zsh
```

Computer is MacOS Catalina (Version 10.15.7).

Here's my VS Code:
![image](https://user-images.githubusercontent.com/7562793/121780466-096a5b80-cb66-11eb-93fc-46550c9c0e8d.png)

**Comment by krzysztofjeziorny** [comment]
Happens to me ([again](https://github.com/microsoft/vscode/issues/123278)) with `fish`, where the integrated terminal sets itself to `zsh` on start.

`"terminal.integrated.defaultProfile.osx": "fish",` 

Also, having set the depracated settings:

```json
  "terminal.integrated.shell.osx": "fish",
  "terminal.integrated.shellArgs.osx": "fish",
```

prompted to migrate them appears in settings:

```json
  "terminal.integrated.profiles.osx": {
    "zsh (migrated)": {
      "path": "zsh",
      "args": ["-l"]
    }
  }
```

Changing the path`zsh` to `fish` (as `fish` or `/usr/local/bin/fish`) doesn't help with the issue.

MacOS 11.5.2, VSCode 1.6.0

**Comment by kondei** [comment]
I have the similar problem.
I set bash as default, but VS Code integrated terminal always starts with zsh...

**Comment by emmahsax** [comment]
I haven't checked on this issue in a while (or to see if it was fixed), but I checked today and noticed that this issue seems to be resolved for me! Here's my current VS Code settings:

```
Version: 1.60.1 (Universal)
Commit: 83bd43bc519d15e50c4272c6cf5c1479df196a4d
Date: 2021-09-10T17:09:14.403Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 19.6.0
```

I'm not sure if the other problems that others mentioned with fish and bash are still an issue, but it could be worth everybody checking again @krzysztofjeziorny and @kondei. I know that my zsh terminals are starting just perfectly now.

I'm going to close this issue, since my issue seems fixed, but if anyone thinks I should reopen it, just write a comment on the issue and tag me and I'll reopen it.
